src/main/java/com/minecolonies/items/ItemScanTool.java

```diff
@@ -67,12 +84,46 @@ else if (!compound.hasKey("pos2"))
             @NotNull BlockPos pos2 = BlockPosUtil.readFromNBT(compound, "pos2");
             if (!worldIn.isRemote)
             {
-                String result = StructureWrapper.saveStructure(worldIn, pos1, pos2);
-                LanguageHandler.sendPlayerMessage(playerIn, result);
+                saveStructure(worldIn, pos1, pos2, playerIn);
             }
             compound.removeTag("pos1");
             compound.removeTag("pos2");
             return EnumActionResult.SUCCESS;
         }
     }
+
+    /**
+     * Scan the structure and save it to the disk.
+     *
+     * @param world Current world.
+     * @param from  First corner.
+     * @param to    Second corner.
+     * @param player causing this action.
+     */
+    private static void saveStructure(@Nullable World world, @Nullable BlockPos from, @Nullable BlockPos to, @NotNull EntityPlayer player)
+    {
+        if (world == null || from == null || to == null)
+        {
+            throw new IllegalArgumentException("Invalid method call, arguments can't be null. Contact a developer.");
+        }
+
+        BlockPos blockpos =
+                new BlockPos(Math.min(from.getX(), to.getX()), Math.min(from.getY(), to.getY()), Math.min(from.getZ(), to.getZ()));
+        BlockPos blockpos1 =
+                new BlockPos(Math.max(from.getX(), to.getX()), Math.max(from.getY(), to.getY()), Math.max(from.getZ(), to.getZ()));
+        BlockPos size = blockpos1.subtract(blockpos).add(1, 1, 1);
+
+        WorldServer worldserver = (WorldServer) world;
+        MinecraftServer minecraftserver = world.getMinecraftServer();
+        TemplateManager templatemanager = worldserver.getStructureTemplateManager();
+
+        String currentMillis = Long.toString(System.currentTimeMillis());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'currentMillis' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)